pkg/intermediate/aggregate.go

```diff
@@ -645,9 +693,122 @@ func (a *AggregationProcess) addFieldsForStatsAggregation(record entities.Record
 			}
 		}
 	}
+
+	// Initialize the throughput elements.
+	var timeStart, timeEnd uint32
+	var byteCount, reverseByteCount uint64
+	timeStart, err := getUnsigned32ValueByIeName(record, "flowStartSeconds")
+	if err != nil {
+		return err
+	}
+	timeEnd, err = getUnsigned32ValueByIeName(record, "flowEndSeconds")
+	if err != nil {
+		return err
+	}
+	byteCount, err = getUnsigned64ValueByIeName(record, "octetTotalCount")
+	if err != nil {
+		return err
+	}
+	reverseByteCount, err = getUnsigned64ValueByIeName(record, "reverseOctetTotalCount")
+	if err != nil {
+		return err
+	}
+	incomingVal := byteCount / (uint64(timeEnd - timeStart))
+	reverseIncomingVal := reverseByteCount / (uint64(timeEnd - timeStart))
+	throughputVals := []uint64{incomingVal, reverseIncomingVal}
+
+	for i, element := range antreaThroughputElements {
+		// add common throughput elements
+		value := throughputVals[i]
+		ie, err := registry.GetInfoElement(element, registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		if err = record.AddInfoElement(entities.NewUnsigned64InfoElement(ie, value)); err != nil {
+			return err
+		}
+		// add source throughput elements
+		value = uint64(0)
+		ie, err = registry.GetInfoElement(antreaSourceThroughputElements[i], registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		if fillSrcStats {
+			value = throughputVals[i]
+		}
+		if err = record.AddInfoElement(entities.NewUnsigned64InfoElement(ie, value)); err != nil {
+			return err
+		}
+		// add destination throughput elements
+		value = uint64(0)
+		ie, err = registry.GetInfoElement(antreaDestinationThroughputElements[i], registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		if fillDstStats {
+			value = throughputVals[i]
+		}
+		if err = record.AddInfoElement(entities.NewUnsigned64InfoElement(ie, value)); err != nil {
+			return err
+		}
+	}
+
+	// Initialize the elements flowEndSecondsFromSource/DestinationNode.
+	if timeEnd != uint32(0) {
+		var srcTime, dstTime uint32
+		if fillSrcStats {
+			srcTime = timeEnd
+		}
+		if fillDstStats {
+			dstTime = timeEnd
+		}
+		ie, err := registry.GetInfoElement("flowEndSecondsFromSourceNode", registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		if err = record.AddInfoElement(entities.NewUnsigned32InfoElement(ie, srcTime)); err != nil {
+			return err
+		}
+		ie, err = registry.GetInfoElement("flowEndSecondsFromDestinationNode", registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		if err = record.AddInfoElement(entities.NewUnsigned32InfoElement(ie, dstTime)); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
+func (a *AggregationProcess) updateFlowEndSecondsFromNodes(incomingRecord, existingRecord entities.Record, ieName string, ieVal uint32) uint32 {
```

Comment:
Is it considering not good to pass the record as a parameter? I added multiple common functions to avoid code repetition, but they all need to have record as a parameter.